charts/argo-workflows/templates/controller/workflow-controller-servicemonitor.yaml

```diff
@@ -0,0 +1,29 @@
+{{- if and (or .Values.controller.metricsConfig.enabled .Values.controller.telemetryConfig.enabled) .Values.controller.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "argo-workflows.controller.fullname" . }}
+  labels:
+    {{- include "argo-workflows.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
+    {{- with .Values.controller.serviceMonitor.additionalLabels }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  endpoints:
+  {{- with .Values.controller.metricsConfig.enabled }}
+    - port: metrics
+      path: {{ . }}
```

Comment:
The path value should be from `.Values.controller.metricsConfig.path` - not the `true` that comes from the `with` expression.  Not sure if `with` really helps here or if it's better going back to the old fashioned `if`.

Reply:
thanks! Agreed, here an `if` is definitely clearer.